src/lib.rs

```diff
@@ -155,3 +179,31 @@ pub(crate) fn extract_zip<P: AsRef<Path>>(source_path: P, unpack_path: P) -> Res
 
     Ok(())
 }
+
+// Returns true if app should be kept active in holochain
+fn keep_app_active(installed_app_id: &str, happs_to_keep: HappIds) -> bool {
+    happs_to_keep.contains(&installed_app_id.to_string())
+        && (installed_app_id.starts_with("core") || installed_app_id.starts_with("self"))
```

Comment:
I don't think we can use the "self" prefix nor do we need it for self-hosted happs.  I think instead we need a regex that matches the hosted-happ installed_app_id pattern which shouldn't be deactivated by this code.